dist/s_function

```diff
@@ -40,6 +40,16 @@ done | python dist/s_function_loop.py |
     egrep '\{@[^@]*(WT_RET[_A-Z]*)\([^@]*(WT_ERR[_A-Z]*)\(.*err:' |
     sed -e 's/^\([^:]*\): *\([^:]*\):.*/\1:\2: mix of returns and jump to the error label within a loop/'
 
+# Early exits from WT_WITH_* macros.
```

Comment:
@ddanderson I wasn't sure if you wanted this check to stay within `s_function`. But to me it seems very useful, let me know if we should remove this.